crypto/sm2/sm2_crypt.c

```diff
@@ -391,3 +391,350 @@ int sm2_decrypt(const EC_KEY *key,
 
     return rc;
 }
+
+#ifndef OPENSSL_NO_CNSM
+/* GM/T003_2012 Defined Key Derive Function */
+int kdf_gmt003_2012(unsigned char *out, size_t outlen, const unsigned char *Z, size_t Zlen, const unsigned char *SharedInfo, size_t SharedInfolen, const EVP_MD *md)
```

Comment:
Is this meant to be an internal function? Then it should be made `static`. Otherwise, it should probably be implemented in terms of the new `EVP_KDF` API.

Reply:
Agreed.